web/html/src/manager/maintenance/calendar/web-calendar.tsx

```diff
@@ -0,0 +1,262 @@
+  const calendarRef = useRef<any>();
+
+  useEffect(() => {
+    getEvents("initial");
+    setCurrentDate(getApi().currentDataManager.data.currentDate);
+    // Hack: Make Fullcalendar button icons show correctly
+    jQuery(".fc-backButton-button").html('<div class="fa fa-angle-left"></div>');
+    jQuery(".fc-nextButton-button").html('<div class="fa fa-angle-right"></div>');
+    jQuery(".fc-skipBackButton-button").html('<div class="fa fa-angle-double-left"></div>')
+      .prop('title', "Skip to the last maintenance window");
+    jQuery(".fc-skipNextButton-button").html('<div class="fa fa-angle-double-right"></div>')
+      .prop('title', "Skip to the next maintenance window");
+  }, []);
+
+  const getApi = () => {
+    return calendarRef.current.getApi();
+  };
+
+  const getEvents = (operation) => {
+    const date = getDate(operation);
+    if (operation === "initial" || needsUpdate(date)) {
+      const endpoint = `/rhn/manager/api/maintenance/events/${operation}/${props.type}/${date}/${props.id}`;
+      return Network.get(endpoint, "application/json").promise
+        .then(events => {
+          setEvents(events.data);
+          navigateTo(operation);
+          props.clearMessages();
+          setCurrentDate(getApi().currentDataManager.data.currentDate);
+        }).catch(props.responseError)
+    }
+    navigateTo(operation);
+  };
+
+  const onSkipBack = () => {
+    // Get first date of currently displayed range
+    const date = getDate("back");
+    // Check if we need to update the events
+    if (!needsUpdate(date) && skipToPrevEvent(date)) {
+      return;
+    }
+    const endpoint = `/rhn/manager/api/maintenance/events/skipBack/${props.type}/${date}/${props.id}`;
+    return Network.get(endpoint, "application/json").promise
+      .then(events => {
+        if (events.data.length === 0) {
+          props.messages(MessagesUtils.info(t("There are no more past maintenance windows")));
+        } else {
+          setEvents(events.data);
+          // Skip to next event that is not in current month
+          const filteredEvents = events.data.filter(event =>
+            moment.parseZone(event.start).month() !== moment(currentDate).month());
+          const lastEvent = moment.parseZone(filteredEvents[filteredEvents.length - 1].start);
+          getApi().gotoDate(lastEvent.format("YYYY-MM-DD"));
+          props.clearMessages();
+          setCurrentDate(getApi().currentDataManager.data.currentDate);
+        }
+      }).catch(props.responseError)
+  };
+
+  const onSkipNext = () => {
+    // Get last date of currently displayed range
+    let date = getDate("next");
+    // Check if we need to update the events
+    if (!needsUpdate(date) && skipToNextEvent(date)) {
+      return;
+    }
+    // If we are in day view and need to update we want to make sure to always use the first day of the next month
+    if (getApi().currentDataManager.data.currentViewType === "timeGridDay" && moment(date).date() !== 1) {
```

Comment:
`moment(date).date !== 1` looks odd, what's the intent here? I might be misunderstanding, but perhaps you meant `moment(...).isValid()`?

Reply:
`moment(date).date()` returns the day of the month. I only want to add one month here if `date` is not already the first of the month. Otherwise we would jump two months ahead in some cases which I want to prevent

Reply:
Ah okay, yeah, off the top of my head I wasn't sure what comes out of it here since I don't use moment that often day to day. I think a good small change here would be to extract it to a variable with a descriptive name a-la `isFirstDayOfMonth` or w/e.